mozetl/clientsdaily/rollup.py

```diff
@@ -30,23 +42,74 @@ def load_main_summary(spark):
 
 def extract_search_counts(frame):
```

Comment:
The [pivot operation](https://databricks.com/blog/2016/02/09/reshaping-data-with-pivot-in-apache-spark.html) might reduce some of the complexity here.  Something along the lines of `groupBy("did").pivot("search_struct.source", SEARCH_ACCESS_POINTS).agg(...)`